### PR TITLE
Fix FlatManifold::normal_vector for triangular faces

### PR DESCRIPTION
**Pull request by peterrum**
... needed in `VectorTools::compute_no_normal_flux_constraints`.

### CODE REVIEW

File: source/grid/manifold.cc [hunk 1]
```diff
@@ -892,26 +894,45 @@ FlatManifold<dim, spacedim>::normal_vector(
...
   while (true)
     {
       Point<spacedim> F;
-      for (const unsigned int v : GeometryInfo<facedim>::vertex_indices())
-        F += face->vertex(v) *
-             GeometryInfo<facedim>::d_linear_shape_function(xi, v);
+      for (const unsigned int v : face->vertex_indices())
+        F +=
+          face->vertex(v) *
+          (hyper_cube ? GeometryInfo<facedim>::d_linear_shape_function(xi, v) :
```
Comment (peterrum) [inline source/grid/manifold.cc:923]: I am not sure why `GeometryInfo` is used here? Isn't it the same as in `FE_Q`?

Comment (kronbichler) [reply source/grid/manifold.cc:923]: Yes, but it is way too expensive to set up an `FE_Q` here just for the purpose of evaluating the shape functions. Note that this is an inner function that is called for many points on each face of the mesh. I think the right approach would be to add a function that can evaluate the linear basis for all reference cell types that we can query similar to `d_linear_shape_function`.

Comment (peterrum) [reply source/grid/manifold.cc]: @kronbichler What do you think about the updated version?

**Review by kronbichler — APPROVED** [review]
Excellent, thank you!

**Pull request merged**